CHANGELOG.md

```diff
@@ -1,36 +1,52 @@
-* BREAKING CHANGE: String setters now throw IllegalArgumentException instead of RealmError for invalid surrogates.
-* BREAKING CHANGE: DynamicRealm.distinct()/distinctAsync() and Realm.distinct()/distinctAsync() now throw IllegalArgumentException instead of UnsupportedOperationException for invalid type or unindexed field.
-* BREAKING CHANGE: All thread local change listeners are now delayed until the next Looper event instead of being triggered when committing.
-* BREAKING CHANGE: Removed RealmConfiguration.getSchemaMediator() from public API which was deprecated in 0.86.0. Please use RealmConfiguration.getRealmObjectClasses() to obtain the set of model classes (#1797).
-* BREAKING CHANGE: Realm.migrateRealm() throws a FileNotFoundException if the Realm file doesn't exist.
-* BREAKING CHANGE: It is now required to unsubscribe from all Realm RxJava observables in order to fully close the Realm (#2357).
+
+### Breaking changes
+
+* BREAKING: Realm has now to be installed as a Gradle plugin.
+* BREAKING: DynamicRealm.executeTransaction() now directly throws any RuntimeException instead of wrapping it in a RealmException (#1682).
+* BREAKING: DynamicRealm.executeTransaction() now throws IllegalArgumentException instead of silently accepting a null Transaction object.
+* BREAKING: String setters now throw IllegalArgumentException instead of RealmError for invalid surrogates.
+* BREAKING: DynamicRealm.distinct()/distinctAsync() and Realm.distinct()/distinctAsync() now throw IllegalArgumentException instead of UnsupportedOperationException for invalid type or unindexed field.
+* BREAKING: All thread local change listeners are now delayed until the next Looper event instead of being triggered when committing.
+* BREAKING: Removed RealmConfiguration.getSchemaMediator() from public API which was deprecated in 0.86.0. Please use RealmConfiguration.getRealmObjectClasses() to obtain the set of model classes (#1797).
+* BREAKING: Realm.migrateRealm() throws a FileNotFoundException if the Realm file doesn't exist.
+* BREAKING: It is now required to unsubscribe from all Realm RxJava observables in order to fully close the Realm (#2357).
+* DEPRECATED: Realm.getInstance(Context). Use Realm.getInstance(RealmConfiguration) or Realm.getDefaultInstance() instead.
+* DEPRECATED: Realm.getTable(Class) which was public because of the old migration API. Use Realm.getSchema() or DynamicRealm.getSchema() instead.
+* DEPRECATED: Realm.executeTransaction(Transaction, Callback) and replaced it with Realm.executeTransactionAsync(Transaction), Realm.executeTransactionAsync(Transaction, OnSuccess), Realm.executeTransactionAsync(Transaction, OnError) and Realm.executeTransactionAsync(Transaction, OnSuccess, OnError).
+
+### Enhancements
+
 * Added support for custom methods, custom logic in accessors, custom accessor names, interface implementation and public fields in Realm objects. (#909)
```

Comment:
I would remove all the "Added" prefixes as well. Bullets might need a slight rewrite for this though